lib/Sema/TypeCheckType.cpp

```diff
@@ -973,28 +985,36 @@ static Type resolveTypeDecl(TypeDecl *typeDecl, SourceLoc loc,
   auto &diags = ctx.Diags;
   auto lazyResolver = ctx.getLazyResolver();
 
+  // Hack: Don't validate nested typealiases if we only need the structural
+  // type.
+  auto prevalidatingAlias = [](TypeDecl *typeDecl, TypeResolution res) {
+    return isa<TypeAliasDecl>(typeDecl)
+        && !typeDecl->hasInterfaceType()
+        && typeDecl->getDeclContext()->isTypeContext()
+        && res.getStage() == TypeResolutionStage::Structural;
+  };
+
   // Don't validate nominal type declarations during extension binding.
-  if (!options.is(TypeResolverContext::ExtensionBinding) ||
-      !isa<NominalTypeDecl>(typeDecl)) {
+  if ((!options.is(TypeResolverContext::ExtensionBinding) ||
+       !isa<NominalTypeDecl>(typeDecl)) &&
+      !prevalidatingAlias(typeDecl, resolution)) {
     // Validate the declaration.
     if (lazyResolver)
       lazyResolver->resolveDeclSignature(typeDecl);
```

Comment:
You could try removing this altogether, and validating the decl only once you know you need the interface type.

Reply:
There's some deep cycles at work that break the standard library if this is removed.